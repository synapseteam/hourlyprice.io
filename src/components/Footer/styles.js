import { css } from "@emotion/react";

export const styles = {
  footer: (theme) => css`
    display: flex;
    justify-content: center;
    align-items: center;
    border-top: ${theme.footerBorderTop};
    padding: 0.5rem;
    background-color: ${theme.quinary};
    box-shadow: ${theme.footerShadow};
    transition: background 0.5s;
    margin-top: auto;
  `,

  footerText: css`
    font-weight: 500;
    font-size: 0.9rem;
  `,
  footerLink: (theme) => css`
    text-decoration: none;
    color: ${theme.senary};
  `,
};
