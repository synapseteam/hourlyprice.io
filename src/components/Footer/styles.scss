$light-purple: #31335a;
$white: #ffffff;

.footer {
  display: flex;
  justify-content: center;
  align-items: center;
  border-top: 1px solid $light-purple;
  padding: 0.6rem;

  .footer-text {
    font-weight: 500;
    font-size: 0.9rem;
  }
}
