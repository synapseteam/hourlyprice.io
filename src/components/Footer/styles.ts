import { css, Theme, SerializedStyles } from "@emotion/react";

export const styles = {
  footer: (theme: Theme): SerializedStyles => css`
    display: flex;
    justify-content: center;
    align-items: center;
    border-top: ${theme.footerBorderTop};
    padding: 0.5rem;
    background-color: ${theme.primary};
    box-shadow: ${theme.footerShadow};
    transition: background 0.5s;
    margin-top: auto;
  `,

  footerText: css`
    font-weight: 500;
    font-size: 0.9rem;
  `,
  footerLink: (theme: Theme): SerializedStyles => css`
    text-decoration: none;
    color: ${theme.senary};
  `,
};
