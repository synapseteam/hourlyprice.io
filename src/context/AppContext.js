import { createContext, useReducer, useContext } from "react";

const AppThemeContext = createContext();

const initState = {
  darkMode: true,
};

const TOGGLE_THEME = "TOGGLE_THEME";

function reducer(state, action) {
  switch (action.type) {
    case TOGGLE_THEME:
      return {
        ...state,
        darkMode: !state.darkMode,
      };
    default:
      throw Error("impossible case while switching Theme");
  }
}

export const AppThemeProvider = ({ children }) => {
  const [state, dispatch] = useReducer(reducer, initState);
  const value = [state, dispatch];

  return (
    <AppThemeContext.Provider value={value}>
      {children}
    </AppThemeContext.Provider>
  );
};

export const useAppThemeContext = () => {
  const context = useContext(AppThemeContext);

  if (!context) {
    throw Error("useAppThemeContext should be used within AppThemeProvider");
  }

  return context;
};

export const toggleTheme = () => {
  return { type: TOGGLE_THEME, payload: null };
};

export default AppThemeContext;
