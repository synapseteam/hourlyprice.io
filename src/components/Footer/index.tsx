/** @jsxImportSource @emotion/react */
import { styles } from "./styles";

interface IProps {
  companyName: string;
  companyUrl: string;
}
const Footer: React.FC<IProps> = ({
  companyName = "Synapse Team LLC",
  companyUrl = "https://synapseteam.com",
}): JSX.Element => {
  const currentYear = new Date().getFullYear();

  return (
    <footer css={styles.footer}>
      <p css={styles.footerText}>
        <a
          css={styles.footerLink}
          href={companyUrl}
          target="_blank"
          rel="noreferrer"
        >{`Â© ${companyName}, ${currentYear}`}</a>
      </p>
    </footer>
  );
};

export default Footer;
