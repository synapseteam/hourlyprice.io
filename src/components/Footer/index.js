import React from "react";
import PropTypes from "prop-types";
import styled from "@emotion/styled";

import { useAppThemeContext } from "context/AppContext";

import {
  lightPurple,
  purple,
  white,
  darkGrey,
} from "components/shared/sharedStylesEmotion/colors.js";

const StyledFooter = styled.footer`
  display: flex;
  justify-content: center;
  align-items: center;
  border-top: 1px solid ${lightPurple};
  border: ${({ darkMode }) => !darkMode && "none"};
  padding: 0.6rem;
  background-color: ${({ darkMode }) => (darkMode ? purple : white)};
  ${({ darkMode }) => !darkMode && "box-shadow: -1px -1px 3px #ddd;"};
`;

const StyledFooterText = styled.p`
  font-weight: 500;
  font-size: 0.9rem;
`;

const StyledLink = styled.a`
  text-decoration: none;
  color: ${({ darkMode }) => (darkMode ? white : darkGrey)};
`;

export default function Footer({ companyName, companyUrl }) {
  const currentYear = new Date().getFullYear();

  const [{ darkMode }] = useAppThemeContext();

  return (
    <StyledFooter darkMode={darkMode}>
      <StyledFooterText>
        <StyledLink
          href={companyUrl}
          target="_blank"
          rel="noreferrer"
          darkMode={darkMode}
        >{`Â© ${companyName}, ${currentYear}`}</StyledLink>
      </StyledFooterText>
    </StyledFooter>
  );
}

Footer.propTypes = {
  companyName: PropTypes.string,
  companyUrl: PropTypes.string,
};

Footer.defaultProps = {
  companyName: "Synapse Team LLC",
  companyUrl: "https://synapseteam.com",
};
