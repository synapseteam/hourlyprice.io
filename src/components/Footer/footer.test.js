import React from "react";
import renderer from "react-test-renderer";
import { AppThemeProvider } from "context/AppContext";

import Footer from "components/Footer/index.js";

if (!global.React) {
  global.React = React;
}

it("Footer rendered correctly with dark Theme", () => {
  const state = {
    darkMode: true,
  };
  const value = [state];
  const tree = renderer
    .create(
      <AppThemeProvider value={value}>
        <Footer
          companyName="Synapse Team LLC"
          companyUrl="https://synapseteam.com"
        />
      </AppThemeProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("Footer rendered correctly with light Theme", () => {
  const state = {
    darkMode: false,
  };
  const value = [state];
  const tree = renderer
    .create(
      <AppThemeProvider value={value}>
        <Footer
          companyName="Synapse Team LLC"
          companyUrl="https://synapseteam.com"
        />
      </AppThemeProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
