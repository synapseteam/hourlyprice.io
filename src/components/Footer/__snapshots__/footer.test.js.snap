// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer rendered correctly 1`] = `
<footer
  className="footer"
>
  <p
    className="footer-text"
  >
    <a
      href="https://synapseteam.com"
      rel="noreferrer"
      target="_blank"
    >
      © Synapse Team LLC, 2021
    </a>
  </p>
</footer>
`;

exports[`Footer rendered correctly with dark Theme 1`] = `
<footer
  className="footer"
>
  <p
    className="footer-text"
  >
    <a
      href="https://synapseteam.com"
      rel="noreferrer"
      target="_blank"
    >
      © Synapse Team LLC, 2021
    </a>
  </p>
</footer>
`;

exports[`Footer rendered correctly with light Theme 1`] = `
<footer
  className="footer"
>
  <p
    className="footer-text"
  >
    <a
      href="https://synapseteam.com"
      rel="noreferrer"
      target="_blank"
    >
      © Synapse Team LLC, 2021
    </a>
  </p>
</footer>
`;
